File: app/src/main/java/nl/acidcats/tumblrlikes/di/AppComponent.java
package nl.acidcats.tumblrlikes.di;

import javax.inject.Singleton;

import dagger.Component;
import nl.acidcats.tumblrlikes.data.services.CacheService;
import nl.acidcats.tumblrlikes.ui.MainActivity;
import nl.acidcats.tumblrlikes.ui.fragments.LoadLikesFragment;
import nl.acidcats.tumblrlikes.ui.fragments.LoginFragment;
import nl.acidcats.tumblrlikes.ui.fragments.PhotoFragment;
import nl.acidcats.tumblrlikes.ui.widgets.PhotoActionDialog;
import nl.acidcats.tumblrlikes.ui.fragments.SetupFragment;

/**
 * Created by stephan on 28/03/2017.
 */

@Singleton
@Component(modules = {AppModule.class, DataModule.class})
public interface AppComponent {
    void inject(MainActivity activity);

    void inject(LoginFragment fragment);

    void inject(LoadLikesFragment fragment);

    void inject(PhotoFragment fragment);

    void inject(CacheService service);

    void inject(PhotoActionDialog photoActionDialog);

    void inject(SetupFragment fragment);
}

File: app/src/main/java/nl/acidcats/tumblrlikes/di/AppComponent.kt
package nl.acidcats.tumblrlikes.di

import dagger.Component
import nl.acidcats.tumblrlikes.data.services.CacheService
import nl.acidcats.tumblrlikes.ui.MainActivity
import nl.acidcats.tumblrlikes.ui.screens.load_likes_screen.LoadLikesFragment
import nl.acidcats.tumblrlikes.ui.screens.login_screen.LoginFragment
import nl.acidcats.tumblrlikes.ui.screens.photo_screen.PhotoFragment
import nl.acidcats.tumblrlikes.ui.screens.settingsscreen.SettingsFragment
import nl.acidcats.tumblrlikes.ui.screens.setup_screen.SetupFragment
import javax.inject.Singleton

/**
 * Created by stephan on 28/03/2017.
 */

@Singleton
@Component(modules = [AppModule::class, DataModule::class])
interface AppComponent {
    fun inject(activity: MainActivity)

    fun inject(fragment: LoginFragment)

    fun inject(fragment: LoadLikesFragment)

    fun inject(fragment: PhotoFragment)

    fun inject(service: CacheService)

    fun inject(fragment: SetupFragment)

    fun inject(fragment: SettingsFragment)
}
